src/anchoredinterval.jl

```diff
@@ -121,8 +121,22 @@ end
 
 ##### ACCESSORS #####
 
-Base.first(interval::AnchoredInterval{P}) where P = min(interval.anchor, interval.anchor+P)
-Base.last(interval::AnchoredInterval{P}) where P = max(interval.anchor, interval.anchor+P)
+function Base.first(interval::AnchoredInterval{P}) where P
+    if P < zero(P)
```

Comment:
I'd write this as a ternary I guess.

Reply:
I thought it was visually too busy with the `+`:
```julia
function Base.first(interval::AnchoredInterval{P}) where P
    P < zero(P) ? interval.anchor + P : interval.anchor
end
```

Reply:
You could alternatively write it as `interval.anchor + P * (P < zero(P))` :stuck_out_tongue:

Reply:
Also, we may want to just store the `extrema` as a tuple in the immutable type rather than performing this check every time.

Reply:
This is more complicated than it first seems see: https://github.com/invenia/Intervals.jl/issues/52

Reply:
I'll go with a ternary statement with parameters around it. @ararslan's suggestion is good but there's an issue we need to solve with `LaxZonedDateTimes` which can result in the answer being wrong with you add a `TimePeriod` of 0...

Reply:
I was joking, really. Using a conditional, be it `if` or ternary, seems better to me.